pythran/pythonic/numpy/fft/rfft.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef PYTHONIC_NUMPY_FFT_RFFT_HPP
+#define PYTHONIC_NUMPY_FFT_RFFT_HPP
+
+#include "pythonic/include/numpy/fft/rfft.hpp"
+#include "pythonic/utils/functor.hpp"
+#include "pythonic/include/utils/array_helper.hpp"
+#include <array>
+#include <map>
+#include "pythonic/types/ndarray.hpp"
+#include "pythonic/__builtin__/None.hpp"
+#include "pythonic/__builtin__/ValueError.hpp"
+#include "pythonic/numpy/swapaxes.hpp"
+#include <string.h>
+#include <math.h>
+#include <vector>
+
+#include <stdio.h>
+#include "pythonic/numpy/fft/fftpack.c"
+
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+    namespace fft
+    {
+        // Aux function
+        template<class T, size_t N>
+        types::ndarray<std::complex<typename std::common_type<T, double>::type>, N> _rfft(types::ndarray<T, N> const &in_array, long NFFT, bool norm)
+        {
+            long i;
+            auto shape = in_array.shape();
+            T* dptr = (T*)in_array.buffer;
+            long npts = shape.back();
+            
+            // Create output array.
+            long out_size = NFFT/2+1;
+            auto out_shape = shape;
+            out_shape.back() = out_size;
+            types::ndarray<std::complex<typename std::common_type<T, double>::type>, N> out_array(out_shape,__builtin__::None);
+            
+//            printf("Shape %d %d N = %d -- Flat size: %d NFFT = %d npts = %d out_size %d\n",
+//                   shape[0],shape[1],N,in_array.flat_size(),NFFT,npts,out_size);
+//            for (i = 0; i < NFFT; i++)
+//                printf("in: %.4f\n",dptr[i]);
+            
+            // Create the twiddle factors. These must be kept from call to call as it's very wasteful to recompute them.
+            // This is from fftpack_litemodule.c
+            static std::map<long, types::ndarray<double, 1>> all_twiddles;
```

Comment:
this is not thread safe, we shoyuld either use a lock or make it thread_private

Reply:
True. I'll look into it.